### ISSUE TITLE
"Development" branch doesn't update after release with `toReleaseBranch` merge strategy

### ISSUE THREAD

**Issue by jeetiss** [algolia/shipjs#786]
I've found new bugs with `toReleaseBranch`  merge strategy, while I had tested #777 issue.

**Describe the bug**
`Development` branch doesn't update after release with `toReleaseBranch` merge strategy

**To Reproduce**
1. setup merge strategy `toReleaseBranch`
2. release new version
3. get outdated version of development branch

**Expected behavior**
development branch should be updated in the end of release, but now `ship.js` push updates to `release` branch instead of `development`

**Links**

- https://github.com/gridsome/eslint-plugin-gridsome/runs/618888461?check_suite_focus=true#step:7:50

- https://github.com/jeetiss/try-shipjs/runs/629670824?check_suite_focus=true#step:6:61

**Comment by jeetiss** [comment]
I create repo with reproducing -> https://github.com/jeetiss/shipjs-issue-786

> In my mind

But, shipjs is working in different way now.

## shipjs prepare
- you were on `develop`
- now on `releases/v1.2.3`
- you merge the pr and a commit is added to `master` (yes, shipjs creates pr to [`master`](https://github.com/jeetiss/shipjs-issue-786/pull/1))

## shipjs trigger
- it's triggered by the new commit in [`master`](https://github.com/jeetiss/shipjs-issue-786/actions/runs/91359685/workflow#L5) branch
- it is [published](https://github.com/jeetiss/shipjs-issue-786/runs/630389446?check_suite_focus=true#step:6:20)
- [git checkout `develop`](https://github.com/jeetiss/shipjs-issue-786/runs/630389446?check_suite_focus=true#step:6:26)
- [git merge `master`](https://github.com/jeetiss/shipjs-issue-786/runs/630389446?check_suite_focus=true#step:6:29)
- [git push origin `master`](https://github.com/jeetiss/shipjs-issue-786/runs/630389446?check_suite_focus=true#step:6:38)  (?? why `master`? we checkout to `develop` branch and should push it)
- so now develop is [outdated](https://github.com/jeetiss/shipjs-issue-786)




ps: I have different names in repo and it can be tricky to understand

- `develop` -> `master`
- `master` -> `release`

**Comment by tyankatsu0105** [comment]
I have a similar issue at [here](https://github.com/tyankatsu0105/eslint-plugin)

My branches strategy is these:

- master => Release branch
- develop => Development branch (default)
- topic branches => From develop branch

https://github.com/tyankatsu0105/eslint-plugin/runs/629421080?check_suite_focus=true

```bash
npm notice package size:  2.7 kB
npm notice unpacked size: 5.9 kB
npm notice shasum:        7021c0afaf9e724111197c642e00675a709c89fe
npm notice integrity:     sha512-piQ6tCuRFgPCd[...]mH/i3IARZ1r/A==
npm notice total files:   6
npm notice
+ @tyankatsu0105/eslint-plugin@0.10.3
› Creating a git tag.
$ git tag v0.10.3
› Pushing to the remote.
$ git checkout develop
Switched to a new branch 'develop'
Branch 'develop' set up to track remote branch 'develop' from 'origin'.
$ git merge master
Updating 707490b..ddbed4f
Fast-forward
 CHANGELOG.md      | 9 +++++++++
 package-lock.json | 2 +-
 package.json      | 2 +-
 3 files changed, 11 insertions(+), 2 deletions(-)
    $ git remote add origin-with-token https://xxx@github.com/tyankatsu0105/eslint-plugin
$ git push origin-with-token master
Everything up-to-date
$ git push origin-with-token v0.10.3
To https://github.com/tyankatsu0105/eslint-plugin
 * [new tag]         v0.10.3 -> v0.10.3
› Creating a release on GitHub repository
› All Finished.
```

I think `git push origin-with-token master` is wrong.
I expect `git push origin-with-token develop`.

So I should run these commands on local every released.

```
git checkout master
git pull  origin master
git checkout develop
git pull origin master
```

**Comment by sarahdayan** [comment]
We the same issue.

According to the documentation, when using the ` toReleaseBranch` strategy, the `trigger` command should:

> 1. run tests.
> 2. release to NPM.
> 3. create a git tag(e.g. v1.4.2).
> 4. merge master back to develop.
> 5. push to git remote.

Step 4 (and 5) are important, because even though `develop` is at the latest, it needs the tagged merge commit that Ship.js makes in `master` so that the next release knows where to start from.

For now, it seems that Ship.js does step 4, but not step 5. Here's an excerpt of the output of our CI (our release branch is `next`):

```txt
$ git checkout next
Switched to a new branch 'next'
Branch next set up to track remote branch next from origin.
$ git merge master
Updating 69b99aa..8709f45
Fast-forward
 CHANGELOG.md   | 3 +++
 package.json   | 2 +-
 src/version.js | 2 +-
 3 files changed, 5 insertions(+), 2 deletions(-)
 create mode 100644 CHANGELOG.md
    $ git remote add origin-with-token https://xxx@github.com/xxx
$ git push origin-with-token master
Everything up-to-date
$ git push origin-with-token v0.1.0-beta.0
To https://github.com/xxx
 * [new tag]         v0.1.0-beta.0 -> v0.1.0-beta.0
› Creating a release on GitHub repository
› All Finished.
```

As you can see, Ship.js does merge `master` into `next`, but it never pushes it to `origin`.

Just in case you need it for debugging, here's what our config looks like:

```js
module.exports = {
  mergeStrategy: {
    toReleaseBranch: {
      next: 'master',
    },
  },
};
```

**Comment by eunjae-lee** [comment]
Thank you all. I got confused. I'm releasing the fix now! :)

**Comment by eunjae-lee** [comment]
Released in 0.18.4 🙂
